src/Compilers/Core/Portable/Diagnostic/TriggerDiagnosticDescriptor.cs

```diff
@@ -18,7 +18,7 @@ public class TriggerDiagnosticDescriptor : DiagnosticDescriptor
         /// <param name="customTags">Optional custom tags for the diagnostic. See <see cref="WellKnownDiagnosticTags"/> for some well known tags.</param>
         public TriggerDiagnosticDescriptor(string id, params string[] customTags)
             : base(id, title: "", messageFormat: "", category: "", defaultSeverity: DiagnosticSeverity.Hidden, isEnabledByDefault: true,
-                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable))
+                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable).Append(WellKnownDiagnosticTags.Trigger))
```

Comment:
It would be nice. But wouldn't that lead me to expect that the type is always preserved / lead me to wonder why this wasn't happening for other descriptors that I create? Isn't it better to be consistent and either always round trip (which we can't do)? If we provide guidance that checking the type for the descriptor is fine if user wants to check for trigger diagnostics, then won't users assume that they can do the same for other descriptors too and be surprised when they find it doesn't work?